flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/InputBuffersMetricsTest.java

```diff
@@ -0,0 +1,180 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition.consumer;
+
+import org.apache.flink.core.fs.CloseableRegistry;
+import org.apache.flink.metrics.MetricGroup;
+import org.apache.flink.runtime.io.network.ConnectionID;
+import org.apache.flink.runtime.io.network.ConnectionManager;
+import org.apache.flink.runtime.io.network.NettyShuffleEnvironment;
+import org.apache.flink.runtime.io.network.NettyShuffleEnvironmentBuilder;
+import org.apache.flink.runtime.io.network.TaskEventDispatcher;
+import org.apache.flink.runtime.io.network.TestingConnectionManager;
+import org.apache.flink.runtime.io.network.buffer.Buffer;
+import org.apache.flink.runtime.io.network.metrics.CreditBasedInputBufferPoolUsageGauge;
+import org.apache.flink.runtime.io.network.metrics.ExclusiveBuffersUsageGauge;
+import org.apache.flink.runtime.io.network.metrics.FloatingBuffersUsageGauge;
+import org.apache.flink.runtime.io.network.partition.InputChannelTestUtils;
+import org.apache.flink.runtime.io.network.partition.ResultPartitionID;
+import org.apache.flink.runtime.io.network.partition.ResultPartitionType;
+import org.apache.flink.runtime.metrics.groups.UnregisteredMetricGroups;
+
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test for input buffer usage related metrics.
+ */
+public class InputBuffersMetricsTest {
+
+	@Test
+	public void testBufferUsageMetrics() throws IOException, InterruptedException {
+
+		CloseableRegistry closeableRegistry = new CloseableRegistry();
+
+		final NettyShuffleEnvironment network = new NettyShuffleEnvironmentBuilder().setIsCreditBased(true).build();
+
+		final SingleInputGate inputGate1 = new SingleInputGateBuilder()
+			.setNumberOfChannels(2)
+			.setIsCreditBased(true)
+			.setResultPartitionType(ResultPartitionType.PIPELINED_BOUNDED)
+			.setupBufferPoolFactory(network)
+			.build();
+
+		final SingleInputGate inputGate2 =  new SingleInputGateBuilder()
+			.setNumberOfChannels(2)
+			.setIsCreditBased(true)
+			.setResultPartitionType(ResultPartitionType.PIPELINED_BOUNDED)
+			.setupBufferPoolFactory(network)
+			.build();
+
+		int buffersPerChannel = 2;
+		int extraNetworkBuffersPerGate = 8;
+
+		try {
+			final ResultPartitionID resultPartitionId1 = new ResultPartitionID();
+			final ResultPartitionID resultPartitionId2 = new ResultPartitionID();
+
+			final ConnectionID connectionId1 = new ConnectionID(new InetSocketAddress("localhost", 5000), 0);
+			final ConnectionID connectionId2 = new ConnectionID(new InetSocketAddress("localhost", 5000), 0);
+
+			RemoteInputChannel remoteInputChannel1 = createUnknownInputChannel(network, inputGate1, resultPartitionId1, 0).toRemoteInputChannel(connectionId1);
```

Comment:
We might use `new InputChannelBuilder().buildRemoteAndSetToGate(inputGate1)` instead, also for the following local channel.